### PR TITLE
Revert Remove __trigger

### PR DESCRIPTION
**Pull request by TheIronDev**

### CODE REVIEW

File: spec/v2/providers/fixtures.ts [hunk 1]
```diff
@@ -1,4 +1,28 @@
...
+
+export const MINIMAL_V1_ENDPOINT: ManifestEndpoint = {
+  availableMemoryMb: RESET_VALUE,
+  ingressSettings: RESET_VALUE,
+  maxInstances: RESET_VALUE,
+  minInstances: RESET_VALUE,
+  serviceAccountEmail: RESET_VALUE,
+  timeoutSeconds: RESET_VALUE,
+  vpc: RESET_VALUE,
+};
```
Comment (taeold) [inline spec/v2/providers/fixtures.ts:25]: Unneeded? These are defined in https://github.com/firebase/firebase-functions/blob/master/spec/fixtures.ts#L25 instead.

File: src/common/config.ts [hunk 2]
```diff
@@ -35,6 +35,15 @@ export function firebaseConfig(): AppOptions | null {
...
   }
+  // Could have Runtime Config with Firebase in it as an ENV value.
+  try {
+    const config = JSON.parse(process.env.CLOUD_RUNTIME_CONFIG);
+    if (config.firebase) {
+      return config.firebase;
+    }
+  } catch (e) {
+    // do nothing
+  }
```
Comment (taeold) [inline src/common/config.ts:46]: also unneeded change.

File: spec/v1/providers/firestore.spec.ts [hunk 3]
```diff
@@ -152,9 +175,9 @@ describe("Firestore Functions", () => {
         .database("myDB")
         .namespace("v2")
         .document("users/{uid}")
-        .onWrite((snap, context) => {
-          expectType<{ uid: string }>(context.params);
-        });
+        .onWrite(() => null);
+
+      expect(cloudFunction.__trigger).to.deep.equal(expectedTrigger(resource, "document.write"));
```
[the remark this answers is not in the document]
Comment (TheIronDev) [reply spec/v1/providers/firestore.spec.ts:181]: Any harm in keeping both in the test?

File: spec/v1/providers/analytics.spec.ts [hunk 4]
```diff
@@ -293,12 +305,12 @@ describe("Analytics Functions", () => {
...
 
   describe("process.env.GCLOUD_PROJECT not set", () => {
-    it("should not throw if __endpoint is not accessed", () => {
+    it("should not throw if __trigger is not accessed", () => {
       expect(() => analytics.event("event").onLog(() => null)).to.not.throw(Error);
     });
 
-    it("should throw when __endpoint is accessed", () => {
-      expect(() => analytics.event("event").onLog(() => null).__endpoint).to.throw(Error);
+    it("should throw when trigger is accessed", () => {
```
[the remark this answers is not in the document]
Comment (TheIronDev) [reply spec/v1/providers/analytics.spec.ts:312]: 👍

File: spec/v2/providers/fixtures.ts (hunk 1 above)
Comment (TheIronDev) [reply spec/v2/providers/fixtures.ts:25]: Oof, didn't see they were moved during conflict resolution.

Are you ok with:

```
export { MINIMAL_V1_ENDPOINT, MINIMAL_V2_ENDPOINT } from "../../fixtures";
```

File: src/common/config.ts (hunk 2 above)
Comment (TheIronDev) [reply src/common/config.ts:46]: 👍

File: spec/v1/providers/firestore.spec.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (taeold) [reply spec/v1/providers/firestore.spec.ts:183]: sgtm. I wanted to make sure that the params bits stayed.

[the remark this answers is not in the document]
Comment (TheIronDev) [reply spec/v1/providers/firestore.spec.ts:183]: 👍 👍  sgtm

File: spec/v1/function-builder.spec.ts [hunk 5]
```diff
@@ -41,7 +40,7 @@ describe("FunctionBuilder", () => {
       .auth.user()
       .onCreate((user) => user);
 
-    expect(fn.__endpoint.region).to.deep.equal(["us-east1"]);
+    expect(fn.__trigger.regions).to.deep.equal(["us-east1"]);
```
[the remark this answers is not in the document]
Comment (TheIronDev) [reply spec/v1/function-builder.spec.ts:43]: Weird I couldn't see this comment in the "Changed files" tab. @_____@

Updated!

File: spec/v1/providers/database.spec.ts [hunk 6]
```diff
@@ -280,448 +322,435 @@ describe("Database Functions", () => {
...
-      database.ref("foo").onDelete((event, context) => {
-        expectType<Record<string, never>>(context.params);
-      });
-      database.ref("foo/{bar}").onDelete((event, context) => {
-        expectType<{ bar: string }>(context.params);
-      });
-      database.ref("foo/{bar}/{baz}").onDelete((event, context) => {
-        expectType<{ bar: string; baz: string }>(context.params);
-      });
-    });
```
[the remark this answers is not in the document]
Comment (TheIronDev) [reply spec/v1/providers/database.spec.ts:294]: !! Updated!

File: spec/v1/providers/database.spec.ts [hunk 7]
```diff
@@ -280,448 +322,435 @@ describe("Database Functions", () => {
...
 
-    it("should return the correct staging instance and path strings if domain is present", () => {
-      const [instance, path] = database.extractInstanceAndPath(
-        "projects/_/instances/foo/refs/bar",
-        "firebaseio-staging.com"
-      );
-      expect(instance).to.equal("https://foo.firebaseio-staging.com");
-      expect(path).to.equal("/bar");
-    });
+describe("extractInstanceAndPath", () => {
```
[the remark this answers is not in the document]
Comment (TheIronDev) [reply spec/v1/providers/database.spec.ts:347]: Yeah... some of these merge conflicts.... they.. yeah.  .__.

Updated!

File: spec/v1/function-builder.spec.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (TheIronDev) [reply spec/v1/function-builder.spec.ts:43]: It was not showing up because of the "large files" not loading automatically, thanks Github. T_T

**Review by taeold — APPROVED** [review]

**Pull request merged**